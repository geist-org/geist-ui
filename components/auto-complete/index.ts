import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
