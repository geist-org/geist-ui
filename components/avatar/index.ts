import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
