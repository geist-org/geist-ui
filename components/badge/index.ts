import Badge from './badge'
import BadgeAnchor from './badge-anchor'

export type BadgeComponentType = typeof Badge & {
  Anchor: typeof BadgeAnchor
}
;(Badge as BadgeComponentType).Anchor = BadgeAnchor

export type { BadgeProps, BadgeTypes } from './badge'
export type { BadgeAnchorProps, BadgeAnchorPlacement } from './badge-anchor'
export default Badge as BadgeComponentType
