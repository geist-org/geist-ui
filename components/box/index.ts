import Box from './box'

export type { BoxProps } from './box'
export default Box
