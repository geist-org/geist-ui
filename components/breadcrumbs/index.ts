import Breadcrumbs from './breadcrumbs'
import BreadcrumbsItem from './breadcrumbs-item'
import BreadcrumbsSeparator from './breadcrumbs-separator'

export type BreadcrumbsComponentType = typeof Breadcrumbs & {
  Item: typeof BreadcrumbsItem
  Separator: typeof BreadcrumbsSeparator
}
;(Breadcrumbs as BreadcrumbsComponentType).Item = BreadcrumbsItem
;(Breadcrumbs as BreadcrumbsComponentType).Separator = BreadcrumbsSeparator

export type { BreadcrumbsProps } from './breadcrumbs'
export type { BreadcrumbsItemProps } from './breadcrumbs-item'
export type { BreadcrumbsSeparatorProps } from './breadcrumbs-separator'
export default Breadcrumbs as BreadcrumbsComponentType
