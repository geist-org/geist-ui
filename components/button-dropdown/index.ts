import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './button-dropdown-item'

type ButtonDropdownType = typeof ButtonDropdown & {
  Item: typeof ButtonDropdownItem
}
;(ButtonDropdown as ButtonDropdownType).Item = ButtonDropdownItem

export type { ButtonDropdownProps, ButtonDropdownTypes } from './button-dropdown'
export type {
  ButtonDropdownItemProps,
  ButtonDropdownItemTypes,
} from './button-dropdown-item'
export default ButtonDropdown as ButtonDropdownType
