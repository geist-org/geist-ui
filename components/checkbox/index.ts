import Checkbox from './checkbox'
import CheckboxGroup from './checkbox-group'

export type CheckboxComponentType = typeof Checkbox & {
  Group: typeof CheckboxGroup
}
;(Checkbox as CheckboxComponentType).Group = CheckboxGroup

export type {
  CheckboxProps,
  CheckboxEvent,
  CheckboxEventTarget,
  CheckboxTypes,
} from './checkbox'
export type { CheckboxGroupProps } from './checkbox-group'
export default Checkbox as CheckboxComponentType
