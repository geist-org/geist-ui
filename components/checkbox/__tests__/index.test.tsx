import React from 'react'
import { mount, render } from 'enzyme'
import { Checkbox } from 'components'

describe('Checkbox', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Checkbox checked={true}>Sydney</Checkbox>)
    expect(() => wrapper.unmount()).not.toThrow()
    const rendered = render(<Checkbox>Sydney</Checkbox>)
    expect(rendered).toMatchSnapshot()
  })

  it('should work with different status', () => {
    const wrapper = mount(
      <div>
        <Checkbox type="secondary" />
        <Checkbox type="success" />
        <Checkbox type="warning" />
        <Checkbox type="error" />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should work correctly with initial value', () => {
    let wrapper = mount(<Checkbox checked={true}>Sydney</Checkbox>)
    let input = wrapper.find('input').getDOMNode()
    expect((input as HTMLInputElement).checked).toBeTruthy()

    wrapper = mount(<Checkbox checked={false}>Sydney</Checkbox>)
    input = wrapper.find('input').getDOMNode()
    expect((input as HTMLInputElement).checked).not.toBeTruthy()

    wrapper = mount(<Checkbox initialChecked>Sydney</Checkbox>)
    input = wrapper.find('input').getDOMNode()
    expect((input as HTMLInputElement).checked).toBeTruthy()

    wrapper = mount(<Checkbox initialChecked={false}>Sydney</Checkbox>)
    input = wrapper.find('input').getDOMNode()
    expect((input as HTMLInputElement).checked).not.toBeTruthy()
  })

  it('should change value after click', () => {
    const Wrapper = () => {
      const [state, setState] = React.useState<string>('state1')

      return (
        <Checkbox initialChecked onChange={() => setState('state2')}>
          {state}
        </Checkbox>
      )
    }
    const wrapper = mount(<Wrapper />)
    const input = wrapper.find('input').at(0)
    input.simulate('change')
    expect(wrapper.find('.text').text()).toContain('state2')
  })

  it('should ignore events when disabled', () => {
    const Wrapper = () => {
      const [state, setState] = React.useState<string>('state1')

      return (
        <Checkbox disabled onChange={() => setState('state2')}>
          {state}
        </Checkbox>
      )
    }
    const wrapper = mount(<Wrapper />)
    const input = wrapper.find('input').at(0)
    input.simulate('change')
    expect(wrapper.find('.text').text()).not.toContain('state2')
  })
})
