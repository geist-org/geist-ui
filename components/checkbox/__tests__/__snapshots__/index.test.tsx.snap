// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "fill": "none",
          "viewBox": "0 0 12 12",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "d": "M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z",
              "stroke": "#666",
            },
            "children": Array [],
            "name": "path",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "d": undefined,
              "stroke": undefined,
            },
            "x-attribsPrefix": Object {
              "d": undefined,
              "stroke": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "style",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "type": "checkbox",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "text",
              },
              "children": Array [
                Node {
                  "data": "Sydney",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [
                  Node {
                    "data": "
          label {
            --checkbox-size: .875rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "style",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "style",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "type": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "style",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "fill": undefined,
          "viewBox": undefined,
          "xmlns": "http://www.w3.org/2000/xmlns/",
        },
        "x-attribsPrefix": Object {
          "fill": undefined,
          "viewBox": undefined,
          "xmlns": "",
        },
      },
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      ",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "style",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "type": "checkbox",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "text",
            },
            "children": Array [
              Node {
                "data": "Sydney",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {},
              "children": Array [
                Node {
                  "data": "
          label {
            --checkbox-size: .875rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "style",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "style",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "fill": "none",
            "viewBox": "0 0 12 12",
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "d": "M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z",
                "stroke": "#666",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "stroke": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "stroke": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "fill": undefined,
            "viewBox": undefined,
            "xmlns": "http://www.w3.org/2000/xmlns/",
          },
          "x-attribsPrefix": Object {
            "fill": undefined,
            "viewBox": undefined,
            "xmlns": "",
          },
        },
        "type": "style",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Node {
        "attribs": Object {
          "type": "checkbox",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "text",
          },
          "children": Array [
            Node {
              "data": "Sydney",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "
          label {
            --checkbox-size: .875rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "style",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "style",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "style",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "fill": "none",
              "viewBox": "0 0 12 12",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z",
                  "stroke": "#666",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "stroke": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "stroke": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "fill": undefined,
              "viewBox": undefined,
              "xmlns": "http://www.w3.org/2000/xmlns/",
            },
            "x-attribsPrefix": Object {
              "fill": undefined,
              "viewBox": undefined,
              "xmlns": "",
            },
          },
          "type": "style",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "type": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "text",
        },
        "children": Array [
          Node {
            "data": "Sydney",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "
          label {
            --checkbox-size: .875rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "style",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "style",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "type": "checkbox",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "style",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "fill": "none",
                "viewBox": "0 0 12 12",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "d": "M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z",
                    "stroke": "#666",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "stroke": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "stroke": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "fill": undefined,
                "viewBox": undefined,
                "xmlns": "http://www.w3.org/2000/xmlns/",
              },
              "x-attribsPrefix": Object {
                "fill": undefined,
                "viewBox": undefined,
                "xmlns": "",
              },
            },
            "type": "style",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "type": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "
          label {
            --checkbox-size: .875rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        ",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "style",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "text",
          },
          "children": Array [
            Node {
              "data": "Sydney",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "type": "checkbox",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {},
              "children": Array [
                Node {
                  "data": "
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "style",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "fill": "none",
                  "viewBox": "0 0 12 12",
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "d": "M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z",
                      "stroke": "#666",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                      "stroke": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                      "stroke": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "fill": undefined,
                  "viewBox": undefined,
                  "xmlns": "http://www.w3.org/2000/xmlns/",
                },
                "x-attribsPrefix": Object {
                  "fill": undefined,
                  "viewBox": undefined,
                  "xmlns": "",
                },
              },
              "type": "style",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "type": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "style",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Checkbox should work correctly with different sizes 1`] = `
"<div><label class=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 12 12\\" fill=\\"none\\"><path d=\\"M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z\\" stroke=\\"#666\\"></path></svg><style>
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      </style><input type=\\"checkbox\\"><span class=\\"text\\">mini</span><style>
          label {
            --checkbox-size: .75rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        </style></label><label class=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 12 12\\" fill=\\"none\\"><path d=\\"M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z\\" stroke=\\"#666\\"></path></svg><style>
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      </style><input type=\\"checkbox\\"><span class=\\"text\\">small</span><style>
          label {
            --checkbox-size: .875rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        </style></label><label class=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 12 12\\" fill=\\"none\\"><path d=\\"M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z\\" stroke=\\"#666\\"></path></svg><style>
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      </style><input type=\\"checkbox\\"><span class=\\"text\\">medium</span><style>
          label {
            --checkbox-size: 1rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        </style></label><label class=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 12 12\\" fill=\\"none\\"><path d=\\"M8.5 0.5H3.5C1.84315 0.5 0.5 1.84315 0.5 3.5V8.5C0.5 10.1569 1.84315 11.5 3.5 11.5H8.5C10.1569 11.5 11.5 10.1569 11.5 8.5V3.5C11.5 1.84315 10.1569 0.5 8.5 0.5Z\\" stroke=\\"#666\\"></path></svg><style>
        svg {
          display: inline-flex;
          width: calc(0.86 * var(--checkbox-size));
          height: calc(0.86 * var(--checkbox-size));
          user-select: none;
          opacity: 1;
          cursor: pointer;
        }
      </style><input type=\\"checkbox\\"><span class=\\"text\\">large</span><style>
          label {
            --checkbox-size: 1.125rem;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            width: auto;
            cursor: pointer;
            opacity: 1;
            height: var(--checkbox-size);
            line-height: var(--checkbox-size);
          }

          .text {
            font-size: var(--checkbox-size);
            line-height: var(--checkbox-size);
            padding-left: calc(var(--checkbox-size) * 0.57);
            user-select: none;
            cursor: pointer;
          }

          input {
            opacity: 0;
            outline: none;
            position: absolute;
            width: 0;
            height: 0;
            margin: 0;
            padding: 0;
            z-index: -1;
            background-color: transparent;
          }
        </style></label></div>"
`;
